gothemis/cell/token_protect_test.go

```diff
@@ -442,6 +449,13 @@ func TestTokenProtectDetectExtendedToken(t *testing.T) {
 }
 
 func TestTokenProtectSwapDataAndToken(t *testing.T) {
+	// FIXME(ilammy, 2020-05-25): avoid capacity allocation panics (T1648)
+	// This tests panics on 32-bit architectures due to "int" overflow.
+	// The implementation needs to check for "int" range when casting "C.size_t".
+	if uint64(^uint(0)) == uint64(^uint32(0)) {
```

Comment:
maybe better to split to two functions which test 32/64 bit versions and place them to separate files with specific [build tags](https://golang.org/pkg/go/build/#hdr-Build_Constraints)? One file will be compiled only if run on 32-bit system, another on 64-bit. Then we will not have such check and if somebody runs tests on their own 32 bit machine, tests will be checked.

Reply:
The problem is that it _fails_ on 32-bit machines and don't want that so we temporarily disable the test. It's still there and will be shown as skipped.

I believe that Themis is a portable library so the test suite should be really be platform-dependent. The same tests should pass *everywhere*, regardless of pointer size, number of registers, endianness, etc. (Except for natural limitations, such as impossibility of handling continuous 3 GB data buffers on 32-bit machines.)

So I'd rather have this ugly wart reminding that it's an ugly wart instead of accepting different test suites for different architectures as a normal thing.